backend/src/auth/auth.service.ts

```diff
@@ -1,4 +1,121 @@
-import { Injectable } from '@nestjs/common'
+import {
+    Injectable,
+    Request,
+    NotFoundException,
+    UnauthorizedException,
+    Body,
+} from '@nestjs/common'
+import { User } from '../types/User'
+import { authenticator } from 'otplib'
+import { UserService } from 'src/user/user.service'
+import { toDataURL } from 'qrcode'
 
 @Injectable()
-export class AuthService {}
+export class AuthService {
+    constructor(private userService: UserService) {}
+
+    async generateTwoFactorAuthenticationSecret(user: User) {
+        const secret = authenticator.generateSecret()
```

Comment:
```suggestion
        const secret : string = authenticator.generateSecret()
```

Add some type for clearity